src/interfaces.ts

```diff
@@ -143,14 +143,21 @@ export enum FormatEnum {
   PRECOMPUTED = 'PRECOMPUTED',
 }
 
-export interface PrecomputedFlag {
+export type BasePrecomputedFlag = {
   flagKey?: string;
   allocationKey: string;
   variationKey: string;
   variationType: VariationType;
-  variationValue: string;
   extraLogging: Record<string, string>;
   doLog: boolean;
+};
+
+export interface PrecomputedFlag extends BasePrecomputedFlag {
+  variationValue: string;
+}
```

Comment:
nit: took me a second to figure out what's the purpose of this type and the base one. Using `ObfuscatedPrecomputedFlag` as a name would make it more obvious

I also like using type aliases to specify encoding:
```suggestion
type Base64 = string;

export interface PrecomputedFlag extends BasePrecomputedFlag {
  variationValue: Base64;
}
```